frontend/src/services/filter/FilterServiceImpl.ts

```diff
@@ -31,21 +42,39 @@ export default class FilterServiceImpl implements FilterService {
     type: string,
     cancellation?: CancellationToken
   ): Promise<NodeTypeFilterModel> {
-    return this.http.get<NodeTypeFilterModel>(
-      '/api/filter/node-type',
-      buildRequest(type),
-      cancellation
-    );
+    let cachedValue = this.cache.nodes.get(type);
+
+    if (cachedValue == null) {
+      cachedValue = new SimpleCachedObservable(() =>
```

Comment:
The cancellation token is passed to the `HTTPService`. This will abort the entire request, altough it may already be nearly completed. If a request is canceled, the requested values will not be thrown away but stored in the cache, so it is not useless. Another approach here is to let the request run, without any cancellation and only simulate the cancellation, that is only rejecting the resulting promise. There is an existing helper method for this in the utilities.

Reply:
True, this was the first approach. I have handeled that correctly in the other implementations. I will implement it there as well.